core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -57,6 +57,20 @@ object DataMirror {
     */
   def hasProbeTypeModifier(x: Data): Boolean = x.probeInfo.nonEmpty
 
+  /** Check if a given `Data`, which may be an `Aggregate` contains a Probe
+    * @param x the `Data` to check
+    * @return `true` if x contains a Probe, `false` otherwise
+    */
+  def containsProbeTypeModifier(x: Data): Boolean = {
+    x match {
+      case a: Aggregate =>
+        a.getElements.foldLeft(false)((accumulator, elem) => accumulator || containsProbeTypeModifier(elem))
+      case d => {
+        hasProbeTypeModifier(d)
+      }
+    }
+  }
```

Comment:
Can this be made tail recursive? Also it would be better to early out rather than using a fold (use `.exists` or `.forall`).